ddtrace/tracer/textmap.go

```diff
@@ -230,6 +240,9 @@ type propagator struct {
 
 func (p *propagator) Inject(spanCtx ddtrace.SpanContext, carrier interface{}) error {
 	switch c := carrier.(type) {
+	// QueryCommentCarrier carriers are only supported by the SQLCommentPropagator
+	case QueryCommentCarrier:
+		return nil
```

Comment:
How about we drop the whole propagator thing and simply implement an `SQLQueryCarrier` which you can then inject everything you need into here? Not as an interface, but simply as a string.
```go
type SQLQueryCarrier string
```
Which you can then define as:
```go
carrier := tracer.SQLQueryCarrier(query)
if err := tracer.Inject(spanContext, carrier); err != nil {
    log.Warn("contrib/database/sql: failed to inject query comments: %v", err)
}
// string(carrier) now holds the query with the comment included
```
Then, we should also implement `Extract` for it and use it in tests, dropping the new method added to the `MockTracer` interface.

Also, a few questions, assumptions I made:
* Why do we care about having multiple modes? It terribly complicates the code, why not just add the dynamic tags every time?
* When is it ever the case that there is no Span ID?

Reply:
> Why do we care about having multiple modes? It terribly complicates the code, why not just add the dynamic tags every time? If you don't need them on the other end, just ignore them.

That's a good question. You're welcome to push back on the answer but the idea is that the static mode can be used in situations where making each query essentially unique by adding dynamic values to the query text can cause negative impacts. A few situations where this happens is with client side caches and also `sqlserver` that generates a new plan in its plan cache whenever anything changes in the query text. The static tags mode is therefore used as a more basic mode of sql comment injection that is safer to use.

The static tagging injection is also relevant to prepared statements because that's one case where dynamic tags don't really make sense. Because executions of prepared statements show up in database activity views with the text of the prepared statement, we wouldn't want the _executions_ of those prepared statements to get the dynamic tags that were used during the _creation_ of the prepared statement. 

> When is it ever the case that there is no Span ID? There will always be a span ID if there is a span context. Otherwise, why not simply inject the span context into the query inside tryTrace after the trace is created, instead of complicating things with tracer.WithSpanID?

Again, good question. I hope I can type a clear explanation here but we can go over this when we chat on Monday otherwise. 

> When is it ever the case that there is no Span ID?

There's no span ID/span context when there's no active trace (which would happen when that trace gets created with the sql operation/span being executed). 

> Why not simply inject the span context into the query inside tryTrace after the trace is created, instead of complicating things with tracer.WithSpanID?

The reason we can't just do the injection in `tryTrace` is because of the combination of two factors:
* `tryTrace` happens _after_ sql operations have been executed because of that thing with `driver.ErrSkip` that we handle by skipping the creation of a span. 
* The injection needs to happen _before_ the operation is executed because the query needs to be injected with the sql comments when it's executed by the database.

Reply:
Got it. Thanks for clearing. Let's talk more in the meeting.

Reply:
@gbbr : I think this might be ready for a review. I'm expecting more comments but hopefully the bigger things look better now. 🙇